mlir_graphblas/src/lib/GraphBLAS/GraphBLASLowerPass.cpp

```diff
@@ -805,82 +805,11 @@ class LowerMatrixMultiplyRewrite : public OpRewritePattern<graphblas::MatrixMult
-                      .Case<FloatType>([&](FloatType type)
-                                       { return rewriter.create<AddFOp>(loc, addBlockArg0, addBlockArg1); });
-    } else if (semiring == "min_plus") {
-      Value cmp = llvm::TypeSwitch<Type, Value>(valueType)
-                      .Case<IntegerType>([&](IntegerType type)
-                                         { return rewriter.create<CmpIOp>(loc, CmpIPredicate::slt, addBlockArg0, addBlockArg1); })
-                      .Case<FloatType>([&](FloatType type)
-                                         { return rewriter.create<CmpFOp>(loc, CmpFPredicate::OLT, addBlockArg0, addBlockArg1); });
-      addResult = rewriter.create<SelectOp>(loc, cmp, addBlockArg0, addBlockArg1);
-    } else {
-      return op.emitError("\"" + semiring + "\" is not a supported semiring.");
-    }
-    rewriter.create<graphblas::YieldOp>(loc, graphblas::YieldKind::ADD, addResult);
-
-    // Insert multiplicative operation
-    Region &multRegion = newMultOp.getRegion(2);
-    Block *multBlock = rewriter.createBlock(&multRegion, {}, {valueType, valueType});
-    Value multBlockArg0 = multBlock->getArgument(0);
-    Value multBlockArg1 = multBlock->getArgument(1);
-    Value multResult;
-
-    if (semiring == "plus_pair") {
-      multResult = llvm::TypeSwitch<Type, Value>(valueType)
-                       .Case<IntegerType>([&](IntegerType type)
-                                          { return rewriter.create<ConstantIntOp>(loc, 1, type.getWidth()); })
-                       .Case<FloatType>([&](FloatType type)
-                                        { return rewriter.create<ConstantFloatOp>(loc, APFloat(1.0), type); });
-    } else if (semiring == "plus_times") {
-      multResult = llvm::TypeSwitch<Type, Value>(valueType)
-                       .Case<IntegerType>([&](IntegerType type)
-                                          { return rewriter.create<MulIOp>(loc, multBlockArg0, multBlockArg1); })
-                       .Case<FloatType>([&](FloatType type)
-                                        { return rewriter.create<MulFOp>(loc, multBlockArg0, multBlockArg1); });
-    } else if (semiring == "plus_plus" || semiring == "min_plus") {
-      multResult = llvm::TypeSwitch<Type, Value>(valueType)
-                       .Case<IntegerType>([&](IntegerType type)
-                                          { return rewriter.create<AddIOp>(loc, multBlockArg0, multBlockArg1); })
-                       .Case<FloatType>([&](FloatType type)
-                                        { return rewriter.create<AddFOp>(loc, multBlockArg0, multBlockArg1); });
-    } else {
-      return op.emitError("\"" + semiring + "\" is not a supported semiring.");
-    }
-    rewriter.create<graphblas::YieldOp>(loc, graphblas::YieldKind::MULT, multResult);
+    LogicalResult result = populateSemiringRegions(rewriter, loc, semiring, valueType, 
```

Comment:
A couple of comments on the WIP, please feel free to ignore them. Here I would simply write: 
``` 
if (failed(populareSemiringsRegion(...)) 
  return ...
```
It saves few lines.